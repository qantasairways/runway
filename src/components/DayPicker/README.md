```js
const disabledBefore = new Date();
const today = new Date();

<StatefulManager initial>
  {({ value, updater }) => (
    <DayPicker
      startDate={value.startDate}
      endDate={value.endDate}
      disabledBefore={disabledBefore}
      disabledAfter={
        new Date(
          disabledBefore.getFullYear(),
          disabledBefore.getMonth(),
          disabledBefore.getDate() + 60
        )
      }
      firstDayOfWeek={1}
      monthsToShow={3}
      startSelectedLabel="SYD"
      endSelectedLabel="MEL"
      Icon={PlaneIcon}
      onDayClick={(startDate, endDate) => updater({ startDate, endDate })}
      isDateRange={true}
      hasPrice={true}
      priceInPoints={false}
      preFooterInfo="Lowest economy price per adult in AUD for a return trip."
      endDateData={{
        price: {
          value: 15127.5,
          isLowestPrice: false,
          points: 10200,
          isClassic: false
        },
        currencySymbol: '$'
      }}
      transformDatesData={arg => {
        return arg.map((date, i) => {
          if (i === 5) {
            console.log(date);
            return {
              ...date,
              currencyCode: 'AUD',
              currencySymbol: '$',
              price: {
                isClassic: true,
                isLowestPrice: true,
                points: 19100,
                taxValue: 29.98,
                value: 126.5
              }
            };
          }
          if (i === 6) {
            console.log(date);
            return {
              ...date,
              currencyCode: 'AUD',
              currencySymbol: '$',
              price: {
                isClassic: null,
                isLowestPrice: true,
                points: 19100,
                taxValue: null,
                value: 126.5
              }
            };
          }
          return date;
        });
      }}
      disclaimerMessage="Please read and agree to the terms and conditions."
    />
  )}
</StatefulManager>;
```
