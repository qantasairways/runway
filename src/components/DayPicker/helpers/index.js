import {
  startOfMonth,
  endOfMonth,
  isBefore,
  isAfter,
  isSameDay,
  addMonths,
  startOfWeek,
  endOfWeek,
  isSameMonth,
  addDays,
  differenceInCalendarMonths,
  isFirstDayOfMonth,
  differenceInWeeks,
  startOfDay
} from 'date-fns';
import {
  KEY_CODE_RIGHT,
  KEY_CODE_LEFT,
  KEY_CODE_DOWN,
  KEY_CODE_UP
} from '../../../constants/keyCodes';

export const MONTH_CAPTION_HEIGHT_MOBILE = 58;
export const DAY_CELL_HEIGHT_MOBILE = 94;
export const DAY_CELL_BORDER_WIDTH = 1;
export const MONTH_CAPTION_HEIGHT_DESKTOP = 78;
export const DAY_CELL_HEIGHT_DESKTOP = 90;

export const isDayBefore = (firstDate, secondDate) => {
  const first = startOfDay(firstDate);
  const second = startOfDay(secondDate);
  return isBefore(first, second);
};

export const isDayAfter = (firstDate, secondDate) => {
  const first = startOfDay(firstDate);
  const second = startOfDay(secondDate);
  return isAfter(first, second);
};

export const getDateWithoutTime = date =>
  date ? new Date(new Date(date).setHours(0, 0, 0, 0)) : null;

export const getTwoDigitDate = date => `0${date.getDate()}`.slice(-2);

export const getMonthAndYear = (date, monthLabels) =>
  `${monthLabels[date.getMonth()]}, ${date.getFullYear()}`;

export const getMonthsArray = (today, monthsToShow) => {
  const monthsArray = [];

  for (let i = 0; i < monthsToShow; i += 1) {
    const month = addMonths(today, i);
    monthsArray.push(month);
  }

  return monthsArray;
};

export function getDateArray({
  month,
  monthIndex,
  today,
  firstDayOfWeek,
  startDate,
  endDate,
  disabledBefore,
  disabledAfter
}) {
  const arr = [];
  const initialDay =
    monthIndex === 0 ? startOfWeek(month) : startOfMonth(month);
  const endDay = endOfWeek(endOfMonth(month));
  const current = startOfWeek(initialDay, { weekStartsOn: firstDayOfWeek });
  const firstAvailableDay = disabledBefore || today;
  const startOfAvailableWeek = startOfWeek(firstAvailableDay, {
    weekStartsOn: firstDayOfWeek
  });
  while (current <= endDay) {
    if (
      !isSameMonth(current, initialDay) ||
      isBefore(current, startOfAvailableWeek)
    ) {
      arr.push({
        date: new Date(current),
        timestamp: current.getTime(),
        isOutside: true
      });
    } else {
      arr.push({
        date: new Date(current),
        timestamp: current.getTime(),
        isToday: isSameDay(current, today),
        isDisabled:
          isBefore(current, disabledBefore) || isAfter(current, disabledAfter),
        isStart: startDate ? isSameDay(current, startDate) : false,
        isEnd: endDate ? isSameDay(current, endDate) : false,
        isFirstDayOfMonth:
          isFirstDayOfMonth(current) || isSameDay(current, disabledBefore),
        isInRange:
          endDate && startDate
            ? isBefore(current, endDate) && isAfter(current, startDate)
            : false
      });
    }
    current.setDate(current.getDate() + 1);
  }
  return arr;
}

export function getInitialDateToFocus(today, startDate, disabledBefore) {
  const dateToFocus = startDate || disabledBefore || today;

  return {
    date: new Date(dateToFocus).getTime(),
    month: differenceInCalendarMonths(dateToFocus, today)
  };
}

export function getItemSize(index, months, firstDayOfWeek, isDesktopDevice) {
  const month = months[index];
  const numberOfWeeks = differenceInWeeks(
    endOfMonth(month),
    index === 0 ? startOfWeek(month) : startOfMonth(month),
    { weekStartsOn: firstDayOfWeek }
  );

  return isDesktopDevice
    ? MONTH_CAPTION_HEIGHT_DESKTOP +
        (DAY_CELL_HEIGHT_DESKTOP + DAY_CELL_BORDER_WIDTH * 3) *
          (numberOfWeeks + 1)
    : MONTH_CAPTION_HEIGHT_MOBILE +
        (DAY_CELL_HEIGHT_MOBILE + DAY_CELL_BORDER_WIDTH * 3) *
          (numberOfWeeks + 1);
}

export function getShouldSelectAsStartDate(
  isSelectingStartDate,
  day,
  startDate
) {
  return isSelectingStartDate || isBefore(day, startDate);
}

export function getEndDateFromStartDate(newStartDate, endDate) {
  return isAfter(newStartDate, endDate) ? null : endDate;
}

export function getDateToNavigate(timestamp, keyCode) {
  switch (keyCode) {
    case KEY_CODE_RIGHT:
      return new Date(addDays(timestamp, 1)).getTime();
    case KEY_CODE_LEFT:
      return new Date(addDays(timestamp, -1)).getTime();
    case KEY_CODE_DOWN:
      return new Date(addDays(timestamp, 7)).getTime();
    case KEY_CODE_UP:
      return new Date(addDays(timestamp, -7)).getTime();
    default:
      return null;
  }
}

export function focusDayCell(timestamp) {
  if (timestamp) {
    const elementToFocus = document.querySelector(`.d${timestamp}`);
    if (elementToFocus) elementToFocus.focus();
  }
}

export const getFirstEnabledMonthDate = ({
  monthDate,
  disabledBefore,
  disabledAfter
}) => {
  const firstDayOfMonth = startOfMonth(monthDate);
  let validDate = firstDayOfMonth;
  if (disabledBefore && isDayBefore(validDate, disabledBefore)) {
    validDate = disabledBefore;
  }
  if (disabledAfter && isDayAfter(validDate, disabledAfter)) {
    validDate = disabledAfter;
  }
  return validDate;
};

export const getLastEnabledMonthDate = ({ monthDate, disabledAfter }) => {
  const lastDayOfMonth = endOfMonth(monthDate);
  let validDate = lastDayOfMonth;
  if (disabledAfter && isDayAfter(validDate, disabledAfter)) {
    validDate = disabledAfter;
  }
  return validDate;
};

export function abbrNum(price) {
  if (price >= 1000000) {
    return `${Math.ceil(price / 1000000)}m`;
  }

  if (price >= 1000) {
    return `${Math.ceil(price / 1000)}k`;
  }

  return Math.ceil(price);
}

export function numberWithCommas(x) {
  return x.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',');
}
