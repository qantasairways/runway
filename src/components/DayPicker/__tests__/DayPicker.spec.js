import React from 'react';
import { shallow } from 'enzyme';
import DayPicker from '..';

const today = new Date(2019, 3, 21, 0, 0, 0, 0);
const end = new Date(2019, 4, 1, 0, 0, 0, 0);
const disabledBefore = new Date(2019, 3, 10, 0, 0, 0, 0);
const hiddenBefore = new Date(2019, 3, 8, 0, 0, 0, 0);
const disabledAfter = new Date(2019, 4, 21, 0, 0, 0, 0);
const CalendarIcon = () => <div>Calendar Icon</div>;

const props = {
  startDate: today,
  endDate: end,
  isDateRange: false,
  hiddenBefore,
  disabledBefore,
  disabledAfter,
  onDayClick: e => e,
  firstDayOfWeek: 4,
  monthsToShow: 6,
  buttonLabel: 'Label',
  placeHolder: 'Placeholder',
  headerLabel: 'header label',
  startSelectedLabel: 'SYD',
  endSelectedLabel: 'MEL',
  startLabel: 'Depart',
  endLabel: 'Return',
  startPlaceholder: 'depart when?',
  endPlaceholder: 'return when?',
  startAriaLabel: 'Selected for departure',
  endAriaLabel: 'Selected for return',
  monthLabels: [
    'Jan',
    'Feb',
    'Mar',
    'Apr',
    'May',
    'Jun',
    'Jul',
    'Aug',
    'Sep',
    'Oct',
    'Nov',
    'Dec'
  ],
  dayLabels: ['Mon', 'Tue'],
  closeAriaLabel: 'Close aria label',
  dialogAriaLabel: 'dialog aria label',
  Icon: CalendarIcon,
  transformDatesData: null,
  configOnMonthsShownSubscription: null,
  footerButtonLabel: 'Confirm',
  preFooterInfo: 'Lowest economy price per adult in AUD for a return trip.',
  bottomFootersummaryLabel: 'From',
  hasPrice: false,
  endDateData: null
};

describe('DayPicker', () => {
  let component;

  it('renders correctly with defaults', () => {
    component = shallow(<DayPicker />);

    expect(component).toMatchSnapshot();
  });

  it('renders DayPicker with props provided', () => {
    component = shallow(<DayPicker {...props} />);

    expect(component).toMatchSnapshot();
  });

  describe('renderHeader()', () => {
    it('renders correctly', () => {
      component = shallow(<DayPicker {...props} />);

      const renderedHeader = shallow(
        component.instance().renderHeader({ closeDialog: () => {} })
      );

      expect(renderedHeader).toMatchSnapshot();
    });
  });

  describe('renderMonth()', () => {
    it('renders correctly', () => {
      component = shallow(<DayPicker {...props} />);

      const renderedMonth = shallow(
        component.instance().renderMonth({ index: 0, style: {} }, true)
      );

      expect(renderedMonth).toMatchSnapshot();
    });
  });

  describe('renderButtonDates()', () => {
    it('does not render if start date is not selected', () => {
      component = shallow(
        <DayPicker {...props} startDate={null} endDate={null} />
      );

      expect(component.instance().renderButtonDates()).toBe(null);
    });

    it('renders correctly when start date is selected', () => {
      component = shallow(
        <DayPicker {...props} startDate={today} endDate={null} />
      );

      const renderedDates = shallow(component.instance().renderButtonDates());

      expect(renderedDates).toMatchSnapshot();
    });

    it('renders correctly when start and end dates are selected', () => {
      component = shallow(<DayPicker {...props} />);

      const renderedDates = shallow(component.instance().renderButtonDates());

      expect(renderedDates).toMatchSnapshot();
    });
  });
});
