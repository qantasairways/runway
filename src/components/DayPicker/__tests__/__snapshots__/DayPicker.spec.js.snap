// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DayPicker renderButtonDates() renders correctly when start and end dates are selected 1`] = `
.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow: hidden;
  max-width: 100%;
}

.emotion-0 {
  font-size: 2.25rem;
  margin: 0 12px 20px 12px;
}

<div
  className="emotion-1"
>
  <ButtonContent
    largeButtonValue="21"
    smallButtonValue="Apr, 2019"
  />
  <div
    className="emotion-0"
  >
    -
  </div>
  <ButtonContent
    largeButtonValue="01"
    smallButtonValue="May, 2019"
  />
</div>
`;

exports[`DayPicker renderButtonDates() renders correctly when start date is selected 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow: hidden;
  max-width: 100%;
}

<div
  className="emotion-0"
>
  <ButtonContent
    largeButtonValue="21"
    smallButtonValue="Apr, 2019"
  />
</div>
`;

exports[`DayPicker renderHeader() renders correctly 1`] = `
.emotion-10 {
  z-index: 1;
}

.emotion-5 {
  font-weight: 700;
  background: #323232;
  color: #fff;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  max-width: 1030px;
  margin: 0 auto;
  padding: 0 0 0 15px;
}

.emotion-0 {
  display: none;
  width: 126px;
}

@media only screen and (min-width:768px) {
  .emotion-0 {
    display: block;
  }
}

.emotion-1 {
  font-size: 1.125rem;
  line-height: 0;
}

@media only screen and (min-width:768px) {
  .emotion-1 {
    padding-right: 71px;
    font-size: 1.375rem;
  }
}

.emotion-3 {
  height: 48px;
  width: 48px;
  padding: 0;
  line-height: 0;
  background: none;
  border: none;
  cursor: pointer;
}

@media only screen and (min-width:768px) {
  .emotion-3 {
    height: 55px;
    width: 55px;
  }
}

.emotion-2 {
  fill: white;
}

.emotion-9 {
  background: #fafafa;
}

.emotion-8 {
  display: grid;
  grid-template-columns: repeat(7,1fr);
  grid-gap: 1px;
  max-width: 1000px;
  margin: 1px auto 0 auto;
  padding: 0 1px;
  width: 100%;
  height: 30px;
  font-size: 0.875rem;
  line-height: 0.875rem;
  text-align: center;
  box-shadow: 0 1px 1px 0 rgba(0,0,0,0.08),0 0 1px 0 rgba(0,0,0,0.04);
}

@media only screen and (min-width:768px) {
  .emotion-8 {
    height: 40px;
  }
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
}

.emotion-6::after {
  content: "";
  position: absolute;
  right: -1px;
  height: 15px;
  border-left: 1px solid #eeeeee;
}

.emotion-6::after:last-child {
  border-left: none;
}

.emotion-6 undefined {
  font-size: 1.125rem;
}

<div
  className="emotion-10"
>
  <div
    className="emotion-5"
  >
    <div
      className="emotion-4"
    >
      <QantasLogo
        className="emotion-0"
      />
      <span
        className="emotion-1"
      >
        header label
      </span>
      <button
        aria-label="Close aria label"
        className="emotion-3"
        onClick={[Function]}
        type="button"
      >
        <CrossIcon
          className="emotion-2"
          height="32px"
          width="32px"
        />
      </button>
    </div>
  </div>
  <Tabs
    Icon={[Function]}
    endDate="Wed May 01 2019"
    endPlaceholder="return when?"
    endSelectedLabel="Return MEL"
    isDateRange={false}
    isSelectingStartDate={true}
    onTabClick={[Function]}
    startDate="Sun Apr 21 2019"
    startPlaceholder="depart when?"
    startSelectedLabel="Depart SYD"
  />
  <div
    className="emotion-9"
  >
    <div
      className="emotion-8"
    >
      <div
        className="emotion-6"
        key="Mon"
      >
        Mon
      </div>
      <div
        className="emotion-6"
        key="Tue"
      >
        Tue
      </div>
    </div>
  </div>
</div>
`;

exports[`DayPicker renderMonth() renders correctly 1`] = `
.emotion-3 {
  padding-top: 2px;
  text-align: center;
  background: #fafafa;
  pointer-events: auto;
}

.emotion-3:last-child {
  padding-bottom: 40px;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 1.375rem;
  height: 58px;
}

@media only screen and (min-width:768px) {
  .emotion-1 {
    height: 78px;
  }
}

.emotion-0 {
  font-weight: 700;
  margin-right: 5px;
}

.emotion-2 {
  display: grid;
  grid-template-columns: repeat(7,1fr);
  grid-gap: 1px;
  max-width: 1000px;
  margin: 1px auto 0 auto;
  padding: 0 1px;
}

<div
  className="emotion-3"
  role="rowgroup"
  style={Object {}}
>
  <div
    className="emotion-1"
  >
    <span
      className="emotion-0"
    >
      May
    </span>
    <span>
      2019
    </span>
  </div>
  <div
    className="emotion-2"
  >
    <Day
      Icon={[Function]}
      currencyCode=""
      currencySymbol=""
      date={2019-05-22T14:00:00.000Z}
      endAriaLabel="Selected for return"
      endDate={2019-04-30T14:00:00.000Z}
      endLabel="Return"
      endSelectedLabel="MEL"
      isDateRange={false}
      isDesktopDevice={true}
      isDisabled={true}
      isEnd={false}
      isFirstDayOfMonth={false}
      isInRange={false}
      isLoadingPrice={false}
      isOutside={false}
      isSelectingStartDate={true}
      isStart={false}
      isToday={false}
      key="1558533600000"
      month="May"
      onDayClick={[Function]}
      onDayNavigate={[Function]}
      price={null}
      startAriaLabel="Selected for departure"
      startDate={2019-04-20T14:00:00.000Z}
      startLabel="Depart"
      startSelectedLabel="SYD"
      timestamp={1558533600000}
      year={2019}
    />
    <Day
      Icon={[Function]}
      currencyCode=""
      currencySymbol=""
      date={2019-05-23T14:00:00.000Z}
      endAriaLabel="Selected for return"
      endDate={2019-04-30T14:00:00.000Z}
      endLabel="Return"
      endSelectedLabel="MEL"
      isDateRange={false}
      isDesktopDevice={true}
      isDisabled={true}
      isEnd={false}
      isFirstDayOfMonth={false}
      isInRange={false}
      isLoadingPrice={false}
      isOutside={false}
      isSelectingStartDate={true}
      isStart={false}
      isToday={false}
      key="1558620000000"
      month="May"
      onDayClick={[Function]}
      onDayNavigate={[Function]}
      price={null}
      startAriaLabel="Selected for departure"
      startDate={2019-04-20T14:00:00.000Z}
      startLabel="Depart"
      startSelectedLabel="SYD"
      timestamp={1558620000000}
      year={2019}
    />
    <Day
      Icon={[Function]}
      currencyCode=""
      currencySymbol=""
      date={2019-05-24T14:00:00.000Z}
      endAriaLabel="Selected for return"
      endDate={2019-04-30T14:00:00.000Z}
      endLabel="Return"
      endSelectedLabel="MEL"
      isDateRange={false}
      isDesktopDevice={true}
      isDisabled={true}
      isEnd={false}
      isFirstDayOfMonth={false}
      isInRange={false}
      isLoadingPrice={false}
      isOutside={false}
      isSelectingStartDate={true}
      isStart={false}
      isToday={false}
      key="1558706400000"
      month="May"
      onDayClick={[Function]}
      onDayNavigate={[Function]}
      price={null}
      startAriaLabel="Selected for departure"
      startDate={2019-04-20T14:00:00.000Z}
      startLabel="Depart"
      startSelectedLabel="SYD"
      timestamp={1558706400000}
      year={2019}
    />
    <Day
      Icon={[Function]}
      currencyCode=""
      currencySymbol=""
      date={2019-05-25T14:00:00.000Z}
      endAriaLabel="Selected for return"
      endDate={2019-04-30T14:00:00.000Z}
      endLabel="Return"
      endSelectedLabel="MEL"
      isDateRange={false}
      isDesktopDevice={true}
      isDisabled={true}
      isEnd={false}
      isFirstDayOfMonth={false}
      isInRange={false}
      isLoadingPrice={false}
      isOutside={false}
      isSelectingStartDate={true}
      isStart={false}
      isToday={false}
      key="1558792800000"
      month="May"
      onDayClick={[Function]}
      onDayNavigate={[Function]}
      price={null}
      startAriaLabel="Selected for departure"
      startDate={2019-04-20T14:00:00.000Z}
      startLabel="Depart"
      startSelectedLabel="SYD"
      timestamp={1558792800000}
      year={2019}
    />
    <Day
      Icon={[Function]}
      currencyCode=""
      currencySymbol=""
      date={2019-05-26T14:00:00.000Z}
      endAriaLabel="Selected for return"
      endDate={2019-04-30T14:00:00.000Z}
      endLabel="Return"
      endSelectedLabel="MEL"
      isDateRange={false}
      isDesktopDevice={true}
      isDisabled={true}
      isEnd={false}
      isFirstDayOfMonth={false}
      isInRange={false}
      isLoadingPrice={false}
      isOutside={false}
      isSelectingStartDate={true}
      isStart={false}
      isToday={false}
      key="1558879200000"
      month="May"
      onDayClick={[Function]}
      onDayNavigate={[Function]}
      price={null}
      startAriaLabel="Selected for departure"
      startDate={2019-04-20T14:00:00.000Z}
      startLabel="Depart"
      startSelectedLabel="SYD"
      timestamp={1558879200000}
      year={2019}
    />
    <Day
      Icon={[Function]}
      currencyCode=""
      currencySymbol=""
      date={2019-05-27T14:00:00.000Z}
      endAriaLabel="Selected for return"
      endDate={2019-04-30T14:00:00.000Z}
      endLabel="Return"
      endSelectedLabel="MEL"
      isDateRange={false}
      isDesktopDevice={true}
      isDisabled={true}
      isEnd={false}
      isFirstDayOfMonth={false}
      isInRange={false}
      isLoadingPrice={false}
      isOutside={false}
      isSelectingStartDate={true}
      isStart={false}
      isToday={true}
      key="1558965600000"
      month="May"
      onDayClick={[Function]}
      onDayNavigate={[Function]}
      price={null}
      startAriaLabel="Selected for departure"
      startDate={2019-04-20T14:00:00.000Z}
      startLabel="Depart"
      startSelectedLabel="SYD"
      timestamp={1558965600000}
      year={2019}
    />
    <Day
      Icon={[Function]}
      currencyCode=""
      currencySymbol=""
      date={2019-05-28T14:00:00.000Z}
      endAriaLabel="Selected for return"
      endDate={2019-04-30T14:00:00.000Z}
      endLabel="Return"
      endSelectedLabel="MEL"
      isDateRange={false}
      isDesktopDevice={true}
      isDisabled={true}
      isEnd={false}
      isFirstDayOfMonth={false}
      isInRange={false}
      isLoadingPrice={false}
      isOutside={false}
      isSelectingStartDate={true}
      isStart={false}
      isToday={false}
      key="1559052000000"
      month="May"
      onDayClick={[Function]}
      onDayNavigate={[Function]}
      price={null}
      startAriaLabel="Selected for departure"
      startDate={2019-04-20T14:00:00.000Z}
      startLabel="Depart"
      startSelectedLabel="SYD"
      timestamp={1559052000000}
      year={2019}
    />
    <Day
      Icon={[Function]}
      currencyCode=""
      currencySymbol=""
      date={2019-05-29T14:00:00.000Z}
      endAriaLabel="Selected for return"
      endDate={2019-04-30T14:00:00.000Z}
      endLabel="Return"
      endSelectedLabel="MEL"
      isDateRange={false}
      isDesktopDevice={true}
      isDisabled={true}
      isEnd={false}
      isFirstDayOfMonth={false}
      isInRange={false}
      isLoadingPrice={false}
      isOutside={false}
      isSelectingStartDate={true}
      isStart={false}
      isToday={false}
      key="1559138400000"
      month="May"
      onDayClick={[Function]}
      onDayNavigate={[Function]}
      price={null}
      startAriaLabel="Selected for departure"
      startDate={2019-04-20T14:00:00.000Z}
      startLabel="Depart"
      startSelectedLabel="SYD"
      timestamp={1559138400000}
      year={2019}
    />
    <Day
      Icon={[Function]}
      currencyCode=""
      currencySymbol=""
      date={2019-05-30T14:00:00.000Z}
      endAriaLabel="Selected for return"
      endDate={2019-04-30T14:00:00.000Z}
      endLabel="Return"
      endSelectedLabel="MEL"
      isDateRange={false}
      isDesktopDevice={true}
      isDisabled={true}
      isEnd={false}
      isFirstDayOfMonth={false}
      isInRange={false}
      isLoadingPrice={false}
      isOutside={false}
      isSelectingStartDate={true}
      isStart={false}
      isToday={false}
      key="1559224800000"
      month="May"
      onDayClick={[Function]}
      onDayNavigate={[Function]}
      price={null}
      startAriaLabel="Selected for departure"
      startDate={2019-04-20T14:00:00.000Z}
      startLabel="Depart"
      startSelectedLabel="SYD"
      timestamp={1559224800000}
      year={2019}
    />
    <Day
      Icon={[Function]}
      currencyCode=""
      currencySymbol=""
      date={2019-05-31T14:00:00.000Z}
      endAriaLabel="Selected for return"
      endDate={2019-04-30T14:00:00.000Z}
      endLabel="Return"
      endSelectedLabel="MEL"
      isDateRange={false}
      isDesktopDevice={true}
      isDisabled={false}
      isEnd={false}
      isFirstDayOfMonth={false}
      isInRange={false}
      isLoadingPrice={false}
      isOutside={true}
      isSelectingStartDate={true}
      isStart={false}
      isToday={false}
      key="1559311200000"
      month="May"
      onDayClick={[Function]}
      onDayNavigate={[Function]}
      price={null}
      startAriaLabel="Selected for departure"
      startDate={2019-04-20T14:00:00.000Z}
      startLabel="Depart"
      startSelectedLabel="SYD"
      timestamp={1559311200000}
      year={2019}
    />
  </div>
</div>
`;

exports[`DayPicker renders DayPicker with props provided 1`] = `
<ButtonWithDialog
  Icon={[Function]}
  buttonLabel="Label"
  closeAriaLabel="Close aria label"
  closeOnBlur={false}
  contentPadding="0"
  dialogAriaLabel="dialog aria label"
  dialogStyles={
    Object {
      "background": "white",
      "boxSizing": "border-box",
      "display": "flex",
      "flexDirection": "column",
      "fontFamily": "\\"Ciutadella\\", sans-serif",
      "height": "100%",
      "label": "runway__dialog",
      "left": 0,
      "overflow": "auto",
      "position": "fixed",
      "top": "100%",
      "transition": "top 300ms ease-in-out",
      "width": "100%",
      "zIndex": 1000,
    }
  }
  hasDialogDimensions={false}
  onBeforeClose={[Function]}
  onBlur={[Function]}
  onClose={[Function]}
  onOpen={[Function]}
  placeHolder="Placeholder"
  renderButtonValue={[Function]}
  renderFooter={[Function]}
  renderHeader={[Function]}
  transitionStyles={
    Object {
      "entered": Object {
        "top": 0,
      },
      "entering": Object {
        "top": "100%",
      },
      "exiting": Object {
        "top": "100%",
      },
    }
  }
>
  <Component />
</ButtonWithDialog>
`;

exports[`DayPicker renders correctly with defaults 1`] = `
<ButtonWithDialog
  Icon={[Function]}
  buttonLabel="When"
  closeAriaLabel="Close dialog"
  closeOnBlur={false}
  contentPadding="0"
  dialogAriaLabel="Select dates"
  dialogStyles={
    Object {
      "background": "white",
      "boxSizing": "border-box",
      "display": "flex",
      "flexDirection": "column",
      "fontFamily": "\\"Ciutadella\\", sans-serif",
      "height": "100%",
      "label": "runway__dialog",
      "left": 0,
      "overflow": "auto",
      "position": "fixed",
      "top": "100%",
      "transition": "top 300ms ease-in-out",
      "width": "100%",
      "zIndex": 1000,
    }
  }
  hasDialogDimensions={false}
  onBeforeClose={[Function]}
  onBlur={[Function]}
  onClose={[Function]}
  onOpen={[Function]}
  placeHolder="When?"
  renderButtonValue={[Function]}
  renderFooter={[Function]}
  renderHeader={[Function]}
  transitionStyles={
    Object {
      "entered": Object {
        "top": 0,
      },
      "entering": Object {
        "top": "100%",
      },
      "exiting": Object {
        "top": "100%",
      },
    }
  }
>
  <Component />
</ButtonWithDialog>
`;
